src/diagnostics/core_diagnostics.jl

```diff
@@ -16,19 +27,160 @@ add_diagnostic_variable!(
     end,
 )
 
-# Relative humidity
+###
+# U velocity (3d)
+###
+add_diagnostic_variable!(
+    short_name = "eastward_wind",
+    long_name = "Eastward Wind",
+    units = "m s^-1",
+    comments = "Eastward (zonal) wind component, a prognostic variable",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        return copy(Geometry.UVector.(cache.ᶜu))
+    end,
+)
+
+###
+# V velocity (3d)
+###
+add_diagnostic_variable!(
+    short_name = "northward_wind",
+    long_name = "Northward Wind",
+    units = "m s^-1",
+    comments = "Northward (meridional) wind component, a prognostic variable",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        return copy(Geometry.VVector.(cache.ᶜu))
+    end,
+)
+
+###
+# W velocity (3d)
+###
+# TODO: may want to convert to omega (Lagrangian pressure tendency) as standard output,
+# but this is probably more useful for now
+add_diagnostic_variable!(
+    short_name = "vertical_wind",
+    long_name = "Vertical Wind",
+    units = "m s^-1",
+    comments = "Vertical wind component, a prognostic variable",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        return copy(Geometry.WVector.(cache.ᶜu))
+    end,
+)
+
+###
+# Temperature (3d)
+###
+add_diagnostic_variable!(
+    short_name = "air_temperature",
+    long_name = "Air Temperature",
+    units = "K",
+    comments = "Temperature of air",
+    compute! = (out, state, cache, time) -> begin
+        # FIXME: Avoid extra allocations when ClimaCore overloads .= for this use case
+        # We will want: out .= integrator.u.c.ρ
+        thermo_params = CAP.thermodynamics_params(cache.params)
+        return copy(TD.air_temperature.(thermo_params, cache.ᶜts))
```

Comment:
We don't need copies here because `TD.air_temperature.` already allocates new memory (same for other functions below)